### PR TITLE
Add Scala Native 0.4.0-M2 support

### PR DESCRIPTION
**Pull request by lolgab**

### CODE REVIEW

File: scalalib/api/src/ZincWorkerApi.scala [hunk 1]
```diff
@@ -85,23 +85,28 @@ object Util {
...
+  def scalaJSNativeBinaryVersion(version: String) = version match {
+      case ScalaJSNativeFullVersion(major, minor, patch, suffix) =>
+        if (suffix != null && suffix.startsWith("-M") && patch == "0")
+          version
+        else if (suffix != null && suffix.startsWith("-RC") && patch == "0")
           s"$major.$minor$suffix"
         else
-          major
+          s"$major.$minor"
+  }
```
[the remark this answers is not in the document]
Comment (lolgab) [reply scalalib/api/src/ZincWorkerApi.scala]: Thank you for your review @sjrd, I added a new commit that should fix this.

**Comment by lefou** [comment]
Travis fails (even after some restarts), so I ran the test locally, but got failures too. Are there some additional requirements?

**Comment by lefou** [comment]
> Are there some additional requirements?

Oh, never mind. I overlooked the setup script. Sorry, I can't reproduce a successful test setup for now. Maybe, others can test/comment? Or you can fix the travis setup?

**Comment by lolgab** [comment]
Thank you @lefou.
I see some flakiness in Scala Native tests, because it seems the logger is trying to read the standard output before it is flushed (or something like that).
By the way it failed one time with the 0.3.9 version (that was already there) and another with the version 0.4.0-M2.
In this PR I added the Scala Native tests in the CI script because they were not there.
I think this PR doesn't change the status quo of test flakiness, but, at least, it doesn't seem to create regressions.

**Comment by lolgab** [comment]
There was a test hanging forever because the Scala Native linker takes a long time (and a lot of memory) to optimise in release mode, when a lot of classes are compiled (in this case scalatest was used). So I deactivated the test in release mode, since the release mode itself is already covered somewhere else.

**Comment by lolgab** [comment]
I rebased the PR on top of current master.
My last commit should have solved the wrong test that was failing before.

**Comment by lolgab** [comment]
✅It is green on Travis 🎉

File: scalanativelib/test/src/HelloNativeWorldTests.scala [hunk 2]
```diff
@@ -169,31 +175,35 @@ object HelloNativeWorldTests extends TestSuite {
...
       val paths = Evaluator.resolveDestPaths(
         helloWorldEvaluator.outPath,
         task.ctx.segments
       )
-      val log = os.read(paths.log)
-      assert(
-        evalCount > 0,
-        log.contains("Scala Native")
-      )
+      val stdout = ammonite.ops.%%("./out")(paths.out / 'dest).out.lines
```
Comment (lefou) [inline scalanativelib/test/src/HelloNativeWorldTests.scala]: I would be great if you could use `os.proc` instead of `ammonite.ops` here.

Comment (lolgab) [reply scalanativelib/test/src/HelloNativeWorldTests.scala]: ✅Done

Comment (lolgab) [reply scalanativelib/test/src/HelloNativeWorldTests.scala]: Added again the test on `evalCount > 0` I had removed by mistake.

**Comment by lolgab** [comment]
> I'd like to have a green Travis-CI result. I'll restart flaky jobs (e.g. those envVars tests fail regularly on travis)

The envVars tests failed again. Can you rerun those jobs?

**Comment by lefou** [comment]
Thank you!

**Comment by lolgab** [comment]
Thank you very much @lefou and @sjrd.
So happy it was merged 🎉!

**Pull request merged**